src/_pytest/python.py

```diff
@@ -105,6 +107,7 @@ def pytest_addoption(parser: Parser) -> None:
     )
 
 
+@hookimpl(tryfirst=True)
```

Comment:
Very questionable, may break users

But without this I cannot get a `pytest_generate_tests` to execute after `pytest.mark.parametrize` is applied. `trylast=True` and `hookwrapper=True` don't work

However, with this `tryfirst=True`, I was able to make `pytest_generate_tests` run before the mark in case of `tryfirst=True`

Reply:
UPD: Now I cannot get the other order to work. Added `test_mark_depends_on_hooks` test that fails, pls help

Reply:
Apparently, ordering is not completely respected for module-local and class-local `pytest_generate_tests`, which is evident from how this hook is [treated in the sources](https://github.com/pytest-dev/pytest/blob/8.3.x/src/_pytest/python.py#L459).

I've changed the tests to use `pytest_generate_tests` from a plugin to get the desired ordering.

As a result, the breaking `@tryfirst` on `pytest_generate_tests` is gone.